File: Sprint8/K/C++/main.cpp
// https://contest.yandex.ru/contest/26131/problems/K/
#include <iostream>
#include <string>

std::string refactor(const std::string &s) {
    const std::string allowed_chars = "bdfhjlnprtvxz";
    std::string result;

    for (char c : s) {
        if (allowed_chars.find(c) != std::string::npos) {
            result.push_back(c);
        }
    }

    return result;
}

int main() {
    std::string a, b;
    std::cin >> a >> b;

    a = refactor(a);
    b = refactor(b);

    if (a.compare(b) == 0) {
        std::cout << 0 << std::endl;
    } else if (a.compare(b) < 0) {
        std::cout << -1 << std::endl;
    } else {
        std::cout << 1 << std::endl;
    }

    return 0;
}

File: Sprint8/K/Rust/src/main.rs
// https://contest.yandex.ru/contest/26131/problems/K/
use std::io::BufRead;

fn refactor(s: &str) -> String {
    let allowed_chars = "bdfhjlnprtvxz";
    s.chars().filter(|&c| allowed_chars.contains(c)).collect()
}

fn main() {
    let mut lines = std::io::stdin().lock().lines();

    let mut a = lines.next().unwrap().unwrap();
    let mut b = lines.next().unwrap().unwrap();

    a = refactor(&a);
    b = refactor(&b);

    match a.cmp(&b) {
        std::cmp::Ordering::Equal => println!("0"),
        std::cmp::Ordering::Less => println!("-1"),
        std::cmp::Ordering::Greater => println!("1"),
    }
}
